core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -225,21 +225,25 @@ sealed abstract class IO[E, A] { self =>
    * }}}
    */
   final def bracket[B](release: A => Infallible[Unit])(use: A => IO[E, B]): IO[E, B] =
-    new IO.Bracket(this, (_: ExitResult[E, B], a: A) => release(a), use)
+    IO.bracket(this) {
```

Comment:
This is the wrong implementation, but it's my fault for feeding you the wrong definition of `IO.bracket`:

```scala
final def bracket[E, A, B](
    acquire: IO[E, A]
  )(release: ExitResult[E, A] => Infallible[Unit])(use: A => IO[E, B]): IO[E, B]
```

Instead we need:

```scala
final def bracket0[E, A, B](
    acquire: IO[E, A]
  )(release: (A, ExitResult[E, B]) => Infallible[Unit])(use: A => IO[E, B]): IO[E, B]
```

In fact, I'd make this change (as `bracket0`), and also keep a simpler `bracket` that can be faster:

```scala
final def bracket[E, A, B](
    acquire: IO[E, A]
  )(release: A => Infallible[Unit])(use: A => IO[E, B]): IO[E, B]
```

This will be faster because there is no need to `.run` the `use` action.